### PR TITLE
[4.0] Add prepared statements for plg_extension_finder

### PR DESCRIPTION
**Pull request by HLeithner**
### Summary of Changes
Updated SQL queries to prepared statements and made some cleanups around the queries.

### Testing Instructions
This Plugin is smartsearch related it manage common words on language installation and uninstall.

* Activate smartsearch
* Install a language with common words
* check the #__finder_terms_common table if the words have been added.
* Uninstall a language with common words and check if the table get cleaned up

### Expected result
Nothing changed.

### CODE REVIEW

File: plugins/extension/finder/finder.php [hunk 1]
```diff
@@ -93,11 +94,18 @@ public function onExtensionAfterUninstall($installer, $eid, $removed)
...
+
 		$query = $db->getQuery(true)
-			->select('element, client_id')
-			->from('#__extensions')
-			->where('extension_id = ' . (int) $eid)
-			->where('type = ' . $db->quote('language'));
+			->select($db->quoteName(['element', 'client_id']))
+			->from($db->quoteName('#__extensions'))
+			->where(
+					[
```
Comment (Quy) [inline plugins/extension/finder/finder.php]: Fix indentation.

File: plugins/extension/finder/finder.php [hunk 2]
```diff
@@ -93,11 +94,18 @@ public function onExtensionAfterUninstall($installer, $eid, $removed)
...
+						$db->quoteName('extension_id') . ' = :eid',
+						$db->quoteName('type') . ' = ' . $db->quote('language')
```
Comment (Quy) [inline plugins/extension/finder/finder.php]: Bind.

Comment (HLeithner) [reply plugins/extension/finder/finder.php]: We query a constant text here so we don't need a prepared variable.

File: plugins/extension/finder/finder.php [hunk 3]
```diff
@@ -93,11 +94,18 @@ public function onExtensionAfterUninstall($installer, $eid, $removed)
...
 		$query = $db->getQuery(true)
-			->select('element, client_id')
-			->from('#__extensions')
-			->where('extension_id = ' . (int) $eid)
-			->where('type = ' . $db->quote('language'));
+			->select($db->quoteName(['element', 'client_id']))
+			->from($db->quoteName('#__extensions'))
+			->where(
+				[
+						$db->quoteName('extension_id') . ' = :eid',
```
Comment (Quy) [inline plugins/extension/finder/finder.php]:
```suggestion
					$db->quoteName('extension_id') . ' = :eid',
```

File: plugins/extension/finder/finder.php [hunk 4]
```diff
@@ -93,11 +94,18 @@ public function onExtensionAfterUninstall($installer, $eid, $removed)
...
+						$db->quoteName('type') . ' = ' . $db->quote('language')
```
Comment (Quy) [inline plugins/extension/finder/finder.php]:
```suggestion
					$db->quoteName('type') . ' = ' . $db->quote('language')
```

File: plugins/extension/finder/finder.php [hunk 5]
```diff
@@ -93,11 +94,18 @@ public function onExtensionAfterUninstall($installer, $eid, $removed)
...
-			->where('type = ' . $db->quote('language'));
+			->select($db->quoteName(['element', 'client_id']))
+			->from($db->quoteName('#__extensions'))
+			->where(
+				[
+					$db->quoteName('extension_id') . ' = :eid',
+					$db->quoteName('type') . ' = ' . $db->quote('language')
+				]
+			)
+			->bind(':eid', (int) $eid, ParameterType::INTEGER);
```
Comment (alikon) [inline plugins/extension/finder/finder.php]:
```suggestion
			->bind(':eid', $eid, ParameterType::INTEGER);
```

**Review by alikon — APPROVED** [review]

**Comment by wilsonge** [comment]
Thanks!

**Pull request merged**